feat(urllib3)!: refactor request hook parameters (#2711)
open-telemetry/opentelemetry-python-contrib: CHANGELOG.md

  ([#2580](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2580))
- Populate `{method}` as `HTTP` on `_OTHER` methods from scope for `asgi` middleware
  ([#2610](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2610))

> - Populate `{method}` as `HTTP` on `_OTHER` methods from scope for `fastapi` middleware

Author: Note: incorrect indentation

  ([#2580](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2580))
- Populate `{method}` as `HTTP` on `_OTHER` methods from scope for `asgi` middleware
  ([#2610](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2610))
- Populate `{method}` as `HTTP` on `_OTHER` methods from scope for `fastapi` middleware
([#2682](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2682))

> - `urllib3`: add `method` and `url` parameters to the request hook

Reviewer: ```suggestion
- `urllib3`: add `method` and `url` parameters to the request hook ([#2711](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2711))
```

> - `urllib3`: add `method` and `url` parameters to the request hook

Reviewer: Nit: If you simply say "add" it wouldn't be a breaking change. I would just say "refactor to include `method` and `url`.

  ([#2580](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2580))
- Populate `{method}` as `HTTP` on `_OTHER` methods from scope for `asgi` middleware
  ([#2610](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2610))
- Populate `{method}` as `HTTP` on `_OTHER` methods from scope for `fastapi` middleware
([#2682](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2682))

> - `urllib3`: the request hook has been refactored: the `headers` and `body` parameters have been replaced by a single `request_info: RequestInfo` parameter that now contains `method` and `url` as well ([#2711](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2711))

Reviewer 2: ```suggestion
- `opentelemetry-instrumentation-urllib3` improve request_hook, replacing `headers` and `body` parameters with a single `request_info: RequestInfo` parameter that now contains the `method` and `url` ([#2711](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2711))
```
